src/util/isArray.ts

```diff
@@ -1 +1 @@
-export const isArray = Array.isArray || (x => x && typeof x.length === 'number');
\ No newline at end of file
+export const isArray = Array.isArray || (<T>(x: any): x is T[] => x && typeof x.length === 'number');
```

Comment:
`<T>(x: any): x is T[] => x` is a bit confusing. Do we need all this complexity?

Isn't just `(x: any) => ...` enough?